mass-update.sh

```diff
@@ -0,0 +1,122 @@
+echo ""
+echo "This script will do the following:"
+echo ""
+echo "- flash all the following hosts with the corresponding firmware files currently present in $FILE_DIR"
+echo "- first flash APs, than core routers based on the naming convention"
+echo "- check the availability of the hosts before and after flashing"
+echo "- remove existing host keys and replace them with the current host keys"
+echo "- make sure that at least 16 MB of RAM are available before performing a sysupgrade"
+echo "- delete the local firmware file from disk after flashing"
+echo ""
+echo "The following files will be processed:"
+for FILE_PATH in $SORTED_FILES; do
+    echo "- $(basename "$FILE_PATH")"
+done
+echo ""
+read -rp "Do you want to proceed [y/N]? " choice
+echo ""
+if [[ ! "$choice" =~ ^[Yy]$ ]]; then
+    echo "Exiting..."
+    exit 0
+fi
+
+# Loop through each file
+for FILE_PATH in $SORTED_FILES; do
+    # Horizontal line to separate iterations
+    echo "----------------------------------------"
+
+    # Extract filename
+    FILENAME=$(basename "$FILE_PATH")
+    echo "Processing file: $FILENAME"
+
+    # Build nodename by omitting the ending
+    NODENAME="${FILENAME%.*}"
+    echo "Nodename: $NODENAME"
+
+    # Build hostname
+    HOSTNAME="$NODENAME.olsr"
+    echo "Hostname: $HOSTNAME"
+
+    # Check if hostname is reachable
+    echo "Checking if $HOSTNAME is reachable..."
+    if ping -c 1 "$HOSTNAME" >/dev/null 2>&1; then
+        echo "Hostname $HOSTNAME is reachable"
+
+        # Removing old host key
+        echo "Removing old host key for $HOSTNAME"
+        ssh-keygen -R "$HOSTNAME" >/dev/null 2>&1
+
+        # Performing ssh-keyscan
+        echo "Performing ssh-keyscan for $HOSTNAME"
+        ssh-keyscan "$HOSTNAME" >> ~/.ssh/known_hosts
+
+        # Check memory on remote host
+        MEMORY=$(ssh "root@$HOSTNAME" "free | awk 'NR==2 {print \$4}'")
+        if [ "$MEMORY" -ge $(( $(stat -c %s "$FILE_PATH") / 1024 + 3072 )) ]; then  # File size in KB + 3 MB
+            echo "Memory on $HOSTNAME is sufficient ($MEMORY KB)"
+
+            # SCP the file
+            echo "Copying $FILENAME to $HOSTNAME:/tmp/"
+            if scp "$FILE_PATH" "root@$HOSTNAME:/tmp/"; then
+                # Debug output: Executing sysupgrade
+                echo "Executing sysupgrade on $HOSTNAME"
+                # shellcheck disable=SC2029
+                ssh "root@$HOSTNAME" "sysupgrade '/tmp/$FILENAME'"
+
+                # Wait for hostname to become unreachable
+                echo "Waiting for $HOSTNAME to become unreachable..."
+                while ping -c 1 "$HOSTNAME" >/dev/null 2>&1; do sleep 1; done
+
+                # Wait for 20 seconds before checking hostname reachability again
+                sleep 20
+
+                # Debug output: Waiting for hostname to become reachable again
+                echo "Waiting for $HOSTNAME to become reachable again..."
+                while ! ping -c 1 "$HOSTNAME" >/dev/null 2>&1; do sleep 1; done
```

Comment:
`ping -c` can be combined with `ping -w` so you don't need the loop

Reply:
The idea of the loop is to be be able to wait for quite some time and to be stuck if the device does not come back. Exiting after a deadline e.g. 120s would of course be better at that point. I don't get how to achieve this with the combination of `-c` and `-w` as `ping -c 1 -w 120 nonexisting.olsr` exits immediately with a "Name or service not known" error.